app/models/user.rb

```diff
@@ -9,9 +11,10 @@ class User < ApplicationRecord
 
   def self.authenticate(email, password)
     user = find_by_email(email)
-    if user && user.password == password
+    if user && BCrypt::Password.new(user.password)==password
```

Comment:
There is a special kind of attack that targets secret and password comparison. Here's a blog post about it: https://blog.sqreen.com/developer-security-best-practices-protecting-against-timing-attacks/

I think in this case it is fine because `BCrypt` overrides its `#==` method. I just want to bring it to your attention in case you haven't come across this type of vulnerabilities!

Reply:
oh, thanks for bringing this t my attention.I didn't know about it